chore(rpc): Remove querying EAP via snql

wmak opened the pull request:
- the useRpc Param used with dataset=spans allows the frontend to choose between querying via rpc or snql
- this should be a no-op since even if the frontend wasn't passing useRpc and hitting eap via snql should work via rpc

wmak committed:
chore(rpc): Remove querying EAP via snql

- the useRpc Param used with dataset=spans allows the frontend to choose
  between querying via rpc or snql
- this should be a no-op since even if the frontend wasn't passing
  useRpc and hitting eap via snql should work via rpc

wmak committed:
ref: remove more useRpc references

DominikB2014 reviewed (approved):
This is nicely timed with

https://github.com/getsentry/sentry/issues/89553

wmak committed:
fix: old import

wmak committed:
fix: missing TOP_EVENTS_DATASETS

wmak committed:
fix: missing param

wmak commented on src/sentry/api/endpoints/organization_events_meta.py:
```diff
@@ -183,10 +183,13 @@ def get(self, request: Request, organization) -> Response:
             return Response({})
 
-        use_rpc = request.GET.get("useRpc", "0") == "1"
+        # TODO: remove useRpc param
+        use_eap = (
+            request.GET.get("useRpc", "0") == "1" or request.GET.get("dataset", None) == "spans"
+        )
```
hell yeah ty

wmak committed:
Merge branch 'master' into wmak/chore/remove-spans-eap-dataset

wmak committed:
Merge branch 'master' into wmak/chore/remove-spans-eap-dataset

wmak committed:
fix: merge conflict

wmak committed:
fix: extra variable

DominikB2014 committed:
Merge remote-tracking branch 'origin/master' into wmak/chore/remove-spans-eap-dataset

DominikB2014 committed:
Merge branch 'master' into wmak/chore/remove-spans-eap-dataset

DominikB2014 committed:
Merge branch 'master' into wmak/chore/remove-spans-eap-dataset

DominikB2014 committed:
Merge branch 'master' into wmak/chore/remove-spans-eap-dataset

DominikB2014 committed:
Merge branch 'master' into wmak/chore/remove-spans-eap-dataset

Zylphrex committed:
Merge branch 'master' into wmak/chore/remove-spans-eap-dataset

Zylphrex committed:
Merge branch 'master' into wmak/chore/remove-spans-eap-dataset

wmak committed:
fix: hanging test

- remove useRpc now i know what this endpoint is doing (inheriting from
  /events which seems sketchy but :Shrug)
